website/project/issue-pr-grading.rst

```diff
@@ -0,0 +1,418 @@
+   the end)
+
+Pro tip: If you include an issue number in a commit message, then it
+will appear automatically in that issue, which will make it easier to
+keep track of what commits are associated with a given issue.
+
+Note: Some tasks may simply not lend themselves to reporting any
+intermediate progress, in which case the closing statement (described
+next) would be enough.
+
+Closing statement
+~~~~~~~~~~~~~~~~~
+
+When a task is done, you should close the corresponding GitHub issue,
+and you should include a comment explaining why this task is done. In
+some cases, this is as simple as stating that the work has been done.
+For example, in Task #1 you could simply state “Implemented the `Circle`
+class, including the area, perimeter, and intersection methods”.
+
+This may seem redundant, but take into account that closing an issue may
+not necessarily mean that the task is done. For example, the following
+could also be valid closing statements for Task #1:
+
+-  “We ultimately decided we don’t need a `Circle` class after all. I’m
+   closing this issue, but will reopen if we decide to revisit this”
+-  “The `Circle` class has been implemented, except for the intersection
+   method, which turned out to be more complex than anticipated. I am
+   closing this issue, and have created a separate task (#5) for
+   implementing circle intersections”
+
+If multiple developers are assigned to the issue, the closing statement
+must also include a clear statement of what work was done by each
+developer
+
+Requirements for Pull Requests
+------------------------------
+
+As in many software projects, the ``main`` branch of our Git
+repository should only contain stable code that compiles, runs, and
+passes all tests. This is enforced by making the main branch a
+*protected* branch, meaning that only the instructors and TAs have the
+ability to directly push to that branch.
+
+On top of that, there is an additional ``dev`` (short for “development”)
+branch that acts as a “pre-main” branch. The ``dev`` branch is also
+protected, and should only contain stable code, but it is used as a
+staging area to integrate code (and uncover potential integration
+issues) before publishing code to ``main``.
+
+This means that, to get code into ChiGame, you must make pull request
+to get code merged into the ``dev`` branch (and the instructor and TAs
+will decide when to merge that code into ``main``).
+
+Because the ``dev`` branch should contain stable code, pull requests to
+``dev`` must be carefully reviewed before they are approved. If you want
+to merge code into ``dev``, your code must be reviewed by (1) someone in
+your team and (2) your senior developer. Please note that the senior
+developer will only do their code review once your teammate has completed
+their review. Additionally, when your code affects code
+owned by another team, you must have an additional reviewer from each
+team whose code is affected.
```

Comment:
Would both reviewers from the teams get points? And do two TAs need to be on the review?

Reply:
I believe its that all reviewers get points and only the one TA who is that teams senior dev needs to contribute.

Reply:
Correct, all reviewers get points. I think for these kind of cross-component pull requests, we would only require a code review from the TA of the team that authored the PR (and only get a code review from the other TA if they actually want to be involved in the PR)